lib/node_modules/@stdlib/string/len/lib/should-break.js

```diff
@@ -0,0 +1,165 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2020 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var CR = require( './constants.js' ).CR;
+var LF = require( './constants.js' ).LF;
+var Control = require( './constants.js' ).Control;
+var Extend = require( './constants.js' ).Extend;
+var RegionalIndicator = require( './constants.js' ).Regional_Indicator;
+var SpacingMark = require( './constants.js' ).SpacingMark;
+var L = require( './constants.js' ).L;
+var V = require( './constants.js' ).V;
+var T = require( './constants.js' ).T;
+var LV = require( './constants.js' ).LV;
+var LVT = require( './constants.js' ).LVT;
+var Prepend = require( './constants.js' ).Prepend;
+var ZWJ = require( './constants.js' ).ZWJ;
+var NotBreak = require( './constants.js' ).NotBreak;
+var BreakStart = require( './constants.js' ).BreakStart;
+var Break = require( './constants.js' ).Break;
+var BreakLastRegional = require( './constants.js' ).BreakLastRegional;
+var BreakPenultimateRegional = require( './constants.js' ).BreakPenultimateRegional;
+var ExtendedPictographic = require( './constants.js' ).Extended_Pictographic;
+
+
+// MAIN //
+
+/**
+* Returns whether a break is allowed between the two given grapheme breaking classes according to UAX #29 3.1.1 Grapheme Cluster Boundary Rules on extended grapheme clusters.
+*
+* @param {number} start - start
+* @param {Array<number>} mid - mid
+* @param {number} end - end
+* @param {number} startEmoji - startEmoji
+* @param {Array<number>} midEmoji - midEmoji
+* @param {number} endEmoji - endEmoji
+* @returns {number} break type
+*/
+function shouldBreak( start, mid, end, startEmoji, midEmoji, endEmoji ) {
+	var prevNonExtendIdx;
+	var regIndRange;
+	var nextEmoji;
+	var allEmoji;
+	var previous;
+	var next;
+	var all;
+	/**
+	* Function expression.
+	* @private
+	* @param {number} c - temp
+	* @returns {boolean} - check if extend
+	*/
+	function checker( c ) {
+		return c === RegionalIndicator;
+	}
+	/**
+	* Function expression.
+	* @private
+	* @param {number} c - temp
+	* @returns {boolean} - check if Extend
+	*/
+	function checkerE( c ) {
+		return c === Extend;
+	}
+	all = [ start ].concat( mid ).concat( [ end ] );
```

Comment:
Minor perf nitpick. Rather than using repeatedly using `concat` which allocates new arrays, consider using a combination of `all = mid.slice().push( end ).unshift( start )`. This should reduce the number of new array allocations from `4` to `2`.